optuna/samplers/_tpe/sampler.py

```diff
@@ -576,25 +620,148 @@ def _get_observation_pairs(
 
 def _split_observation_pairs(
     config_vals: Dict[str, List[Optional[float]]],
-    loss_vals: List[Tuple[float, float]],
+    loss_vals: List[Tuple[float, List[float]]],
     n_below: int,
-) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+    weights: Callable[[int], np.ndarray],
+) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray], np.ndarray]:
```

Comment:
IMO, making this func return `weights_below` is somewhat confusing because the name of this function only contains the verb **split**.
How about renaming this func or moving the `weights_below` calculation code into another func?

Reply:
Yes, I agree with you. But, it seems that we cannot move the `weights_below` calculation here. It is because in the multi-objective case we use the `indices_below` computed on the way to compute `below`. We have 2 options here. 
- Rename the function like `_split_observations_pairs_and_calculate_weights_below`
- Make this function a method of TPESampler, preserve `indices_below` in the call of `_split_observation_pairs`, and use it in the another method like `_calculate_weights_below_for_multi_objective`.

Now I'm leaning towards the second option. What do you think?

Reply:
Thank you for your explanation and for proposing options.

Deciding which is the better solution seems a bit difficult issue to me, because,
- the first option will introduce a too long name
- for the second option, will introduce an implicit dependency between methods and could make it a bit harder to read and maintain the code

Alternatively, I'm not sure this is a better option, how about making `_split_observation_pairs` return `indices_below` and `indices_aboce` to make them available on the caller side?

Reply:
Thank you for your proposal. Sounds good to me. Returning not `below` and `above` but `indices_below` and `indices_above` will simplify the codes. Thanks!